From bda9cd4cd090d9d7421d1ee95a5bd4bdeca3b943 Mon Sep 17 00:00:00 2001
From: Yael Samet <yael.samet@intel.com>
Date: Sun, 15 Jan 2017 10:51:05 +0200
Subject: [PATCH 4663/4706] mei: dal: remove redundant code from
 bhp_deinit_internal

commit e883646dfbbe694527880fc2a47e7529020d3bce from
git://git.yoctoproject.org/linux-yocto-4.1

Remove duplicated check for bhp_is_initialized().

Change-Id: Ibf41ebb42e60ced1a2eea3ec8c01d62fbbfd1fc0
Signed-off-by: Yael Samet <yael.samet@intel.com>
Signed-off-by: Tomas Winkler <tomas.winkler@intel.com>
---
 drivers/misc/mei/dal/bhp_impl.c |    7 +------
 1 files changed, 1 insertions(+), 6 deletions(-)

diff --git a/drivers/misc/mei/dal/bhp_impl.c b/drivers/misc/mei/dal/bhp_impl.c
index 268ad24..ac7163f 100644
--- a/drivers/misc/mei/dal/bhp_impl.c
+++ b/drivers/misc/mei/dal/bhp_impl.c
@@ -620,11 +620,6 @@ int bhp_init_internal(void)
 
 int bhp_deinit_internal(void)
 {
-	int ret = 0;
-
-	if (!bhp_is_initialized())
-		return 0;
-
 	mutex_enter(bhm_gInit);
 
 	if (bhp_is_initialized()) {
@@ -635,5 +630,5 @@ int bhp_deinit_internal(void)
 
 	mutex_exit(bhm_gInit);
 
-	return ret;
+	return 0;
 }
-- 
1.7.5.4

